targets/TARGET_STM/ospi_api.c

```diff
@@ -34,7 +34,7 @@
 
 /* Max amount of flash size is 4Gbytes */
 /* hence 2^(31+1), then FLASH_SIZE_DEFAULT = 1<<31 */
-#define OSPI_FLASH_SIZE_DEFAULT 0x80000000
+#define OSPI_FLASH_SIZE_DEFAULT 0x4000000 //512Mbits
```

Comment:
Don't understand why driver is using a "default" flash size...?
All octo SPI are the same?
(maybe comment above is wrong now?)

Reply:
It is copy from qspi_api.c.
I found a issue about this value.
I modified the code below.
`obj->handle.Init.DeviceSize = POSITION_VAL(OSPI_FLASH_SIZE_DEFAULT) - 1;`

`obj->handle.Init.DeviceSize = 32;`

The driver need to execute command RDCR2 to read the bank status of flash. This command need to send address(0xC0000000> OSPI_FLASH_SIZE_DEFAULT). The controller will report error when i execute RDCR2.
SO i need to change the value of  DeviceSize.